src/api/base.js

```diff
@@ -27,10 +28,19 @@ export class BaseApi {
      * @return {string} - an URL
      */
     getFullURL(...args) {
-        const fragments = args.map((fragment) => encodeURIComponent(fragment));
+        const fragments = args.map((a) => encodeURIComponent(a));
         return urljoin(this.serviceUrl, ...fragments);
     }
 
+    getFullURLWithApp(...args) {
+        if (this.requireAppId) {
```

Comment:
could you put a comment about this logic here? I get it, but it's a bit obscure.